### PR TITLE
chore(discoveryManager): expose Discoverer refresh function 

### PR DESCRIPTION
**Pull request by secustor**
This PR exposes the `allGroups` function to consumers of the package.    

Signed-off-by: secustor <sebastian@poxhofer.at>

### CODE REVIEW

**Review by jaronoff97 — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks for your pull request. This function has a side effet on metrics exposed and also returns possibly empty groups. I'd say it's mostly an internal function that we should not expose.

If you need a way to list targets from the manager, we could probably expose another ad-hoc function. What is your usecase here? Are you also using scrape or notify manager?

**Comment by secustor** [comment]
Thanks for your response.   

We are configuring a `discovery.Manager` and combine it with a a scrape manager using a sync channel.     
Basically we are configuring a discovery manager based on a HTTP endpoint which returns links to HTTP_SD endpoints.
     
To test this efficiently, without mocking every Prometheus target, we would like to retrieve the applied HTTP_SD configs and targetgroups.
https://github.com/secustor/opentelemetry-collector-contrib/blob/implement_prometheus_receiver_target_allocator/receiver/prometheusreceiver/metrics_receiver.go#L120

**Comment by roidelapluie** [comment]
Would it be helpful to expose Refresh instead? https://github.com/prometheus/prometheus/blob/0b41fd6e716f3e8b79016f73b6dd77fdf5fcdced/discovery/refresh/refresh.go#L114 that seems closer to what you try to achieve.

**Comment by secustor** [comment]
If I'm would require to expose/create getters for following attributes:
- `discovery.Manager.providers[]`
- `discovery.Provider`
- `discovery.Provider.d` ( Discoverer ) 
- `discovery.http.Discovery.refresh`

If this would be fine you, I would change this PR accordingly.

**Comment by secustor** [comment]
@roidelapluie I have implemented the changes. Can you take another look?     
Thanks in advance!

File: discovery/manager.go [hunk 1]
```diff
@@ -163,6 +168,11 @@ type Manager struct {
 	lastProvider uint
 }
 
+// Providers returns the currently configured SD providers
```
Comment (roidelapluie) [inline discovery/manager.go:171]:
```suggestion
// Providers returns the currently configured SD providers.
```

**Comment by roidelapluie** [comment]
have you validated this downstream?

**Comment by secustor** [comment]
Not yet, but I will post here an update as I can confirm it.

**Comment by secustor** [comment]
@roidelapluie Sry for the delay. 

I can confirm, that I have now validated this downstream.

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**